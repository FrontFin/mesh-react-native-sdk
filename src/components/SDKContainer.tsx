import React from 'react';
import type { ReactNode } from 'react';
import { SafeAreaView, StatusBar } from 'react-native';
import { useSDKStyles } from './SDKContainer.styled';

type Props = {
  children: ReactNode
}

export const SDKContainer = ({ children }: Props) => {
  const { statusBarStyle } = useSDKStyles();

  return (
    <SafeAreaView style={{ flex: 1 }} testID={'link-connect-component'}>
      <StatusBar backgroundColor={statusBarStyle.backgroundColor} />
      {children}
    </SafeAreaView>
  );
}

