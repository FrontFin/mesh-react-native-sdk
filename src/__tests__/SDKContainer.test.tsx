import React from 'react';
import { render } from '@testing-library/react-native';

import { mockedUseColorScheme } from './__mocks__/useColorScheme';

import SDKContainer from '../components/SDKContainer';

describe('SDKContainer', () => {
  mockedUseColorScheme.mockReturnValue('light');

  test('renders correctly', () => {
    const component = render(<SDKContainer>Test Content</SDKContainer>);
    const { getByTestId } = component;

    expect(getByTestId('link-connect-component')).toBeTruthy();

    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();

  });
});
