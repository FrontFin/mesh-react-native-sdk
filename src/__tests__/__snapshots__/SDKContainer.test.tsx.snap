// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SDKContainer renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
  testID="link-connect-component"
>
  Test Content
</RCTSafeAreaView>
`;
